tuxemon/core/states/combat/combat.py

```diff
@@ -694,6 +710,9 @@ def end_combat(self):
         fadeout_action.parameters = [1000]
         event_engine.actions["fadeout_music"]["method"](self.game, fadeout_action)
 
+        #Round player movement in case battle started in the middle of a step
+        self.game.player1.tile_pos = (round(self.game.player1.tile_pos[0]),round(self.game.player1.tile_pos[1]))
```

Comment:
I was doing some testing and still found the player being offset from the tile grid when getting into random battles because of this.

It looks like you actually need to round the player's pixel position (instead of tile position) to the nearest multiple of the base tile size. The reason you have to do this is because the [player's tile position is actually derived from the player's pixel position](https://github.com/Tuxemon/Tuxemon/blob/development/tuxemon/core/states/world/worldstate.py#L297) (thus it gets overwritten every frame).

We might need to do something like this instead here:
```python
self.game.player1.position[0] = int(prepare.TILE_SIZE[0] * round(float(self.game.player1.position[0])/prepare.TILE_SIZE[0]))
self.game.player1.position[1] = int(prepare.TILE_SIZE[1] * round(float(self.game.player1.position[1])/prepare.TILE_SIZE[1]))
```

I have this same calculation in `core.components.map` called ["round_to_divisible"](https://github.com/Tuxemon/Tuxemon/blob/development/tuxemon/core/components/map.py#L570) that is used there a few times. We might be able to move that function into a more generic utility module if it's used in more places.